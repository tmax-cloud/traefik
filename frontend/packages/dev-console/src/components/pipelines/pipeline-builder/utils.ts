import { history, resourcePathFromModel } from '@console/internal/components/utils';
import { apiVersionForModel } from '@console/internal/module/k8s';
import { ClusterTaskModel, PipelineModel } from '@console/internal/models';
import { Pipeline, PipelineResourceTask, PipelineResourceTaskParam, PipelineTask } from '../../../utils/pipeline-augment';
import { getTaskParameters } from '../resource-utils';
import { TASK_ERROR_STRINGS, TaskErrorType } from './const';
import { PipelineBuilderFormikValues, PipelineBuilderFormValues, TaskErrorMap } from './types';

export const getErrorMessage = (errorTypes: TaskErrorType[], errorMap: TaskErrorMap) => (taskName: string): string => {
  if (!taskName) {
    return TASK_ERROR_STRINGS[TaskErrorType.MISSING_NAME];
  }

  const errorList: TaskErrorType[] | undefined = errorMap?.[taskName];
  if (!errorList) return null;

  const hasRequestedError = errorList.filter(error => errorTypes.includes(error));
  return hasRequestedError.length > 0 ? TASK_ERROR_STRINGS[hasRequestedError[0]] : null;
};

export const taskParamIsRequired = (param: PipelineResourceTaskParam): boolean => {
  return false;
};

export const convertResourceToTask = (resource: PipelineResourceTask, runAfter?: string[]): PipelineTask => {
  return {
    name: resource.metadata.name,
    runAfter,
    taskRef: {
      kind: resource.kind === ClusterTaskModel.kind ? ClusterTaskModel.kind : undefined,
      name: resource.metadata.name,
    },
    params: getTaskParameters(resource).map(param => ({
      name: param.name,
      value: param.default,
      type: param.type
    })),
  };
};

export const getPipelineURL = (namespace: string) => {
  return `/k8s/ns/${namespace}/pipelines`;
};

const removeListRunAfters = (task: PipelineTask, listIds: string[]): PipelineTask => {
  if (task?.runAfter && listIds.length > 0) {
    // Trim out any runAfters pointing at list nodes
    const runAfter = (task.runAfter || []).filter(runAfterName => !listIds.includes(runAfterName));

    return {
      ...task,
      runAfter,
    };
  }

  return task;
};

const removeEmptyDefaultParams = (task: PipelineTask): PipelineTask => {
  if (task.params?.length > 0) {
    // Since we can submit, this param has a default; check for empty values and remove
    // 20210929 Policy changed
    task.params.forEach(param => { if (param.value === undefined) { param.value = (param.type && param.type === 'array') ?  [''] : ''; }; delete param.type; });
    return {
      ...task,
      //params: task.params.filter(param => !!param.value),
    };
  }

  return task;
};

export const convertBuilderFormToPipeline = (formValues: PipelineBuilderFormikValues, namespace: string, existingPipeline?: Pipeline): Pipeline => {
  const { name, resources, params, workspaces, tasks, listTasks, metadata } = formValues;
  const listIds = listTasks.map(listTask => listTask.name);
  const labels = metadata?.labels;

  return {
    ...existingPipeline,
    apiVersion: apiVersionForModel(PipelineModel),
    kind: PipelineModel.kind,
    metadata: {
      ...existingPipeline?.metadata,
      name,
      namespace,
      labels,
    },
    spec: {
      ...existingPipeline?.spec,
      params,
      resources,
      workspaces,
      tasks: tasks.map(task => removeEmptyDefaultParams(removeListRunAfters(task, listIds))),
    },
  };
};

export const convertPipelineToBuilderForm = (pipeline: Pipeline): PipelineBuilderFormValues => {
  if (!pipeline) return null;

  const {
    metadata: { name },
    spec: { params = [], resources = [], workspaces = [], tasks = [] },
  } = pipeline;

  return {
    name,
    params,
    resources,
    workspaces,
    tasks,
    listTasks: [],
  };
};

export const goToYAML = (existingPipeline?: Pipeline, namespace?: string) => {
  history.push(existingPipeline ? `${resourcePathFromModel(PipelineModel, existingPipeline?.metadata?.name, existingPipeline?.metadata?.namespace)}/yaml` : `${getPipelineURL(namespace)}/~new`);
};
