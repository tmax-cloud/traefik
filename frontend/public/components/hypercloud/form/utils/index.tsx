export * from './modal-launcher';
export * from './modal-list';
