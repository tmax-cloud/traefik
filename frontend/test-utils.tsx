import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './jest-setup';

// MEMO : redux Provider로 감싼 구조로 @testing-library/react의 render함수를 래핑함.
const render = (ui: React.ReactElement, options?) => {
  const Wrapper = ({ children }: { children?: React.ReactNode }) => {
    return <Provider store={store}>{!!options?.wrapper ? options.wrapper({ children: children }) : children}</Provider>;
  };

  // MEMO : options로 들어온 wrapper는 Wrapper컴포넌트에서 사용하고, 여기서 wrapper속성을 Wrapper로 덮어씌워서 설정해 줌.
  return rtlRender(ui, { ...options, wrapper: Wrapper });
};

export * from '@testing-library/react';
export { render };
