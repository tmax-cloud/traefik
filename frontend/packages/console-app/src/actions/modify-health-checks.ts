import * as _ from 'lodash';
import { K8sKind, K8sResourceKind, referenceFor } from '@console/internal/module/k8s';
import { KebabOption } from '@console/internal/components/utils';

const healthChecksAdded = (resource: K8sResourceKind): boolean => {
  const containers = resource?.spec?.template?.spec?.containers;
  return _.every(containers, container => container.readinessProbe || container.livenessProbe || container.startupProbe);
};

const healthChecksUrl = (model: K8sKind, obj: K8sResourceKind): string => {
  const {
    metadata: { name, namespace },
  } = obj;
  const resourcePlural = model.crd ? referenceFor(obj) : model.plural;
  const containers = obj?.spec?.template?.spec?.containers;
  const containerName = containers?.[0]?.name;
  return `/k8s/ns/${namespace}/${resourcePlural}/${name}/containers/${containerName}/health-checks`;
};

export const AddHealthChecks = (model: K8sKind, obj: K8sResourceKind): KebabOption => {
  return {
    label: 'COMMON:MSG_MAIN_ACTIONBUTTON_14',
    hidden: healthChecksAdded(obj),
    href: healthChecksUrl(model, obj),
    accessReview: {
      group: model.apiGroup,
      resource: model.plural,
      name: obj.metadata.name,
      namespace: obj.metadata.namespace,
      verb: 'update',
    },
  };
};

export const EditHealthChecks = (model: K8sKind, obj: K8sResourceKind): KebabOption => {
  return {
    label: 'COMMON:MSG_MAIN_ACTIONBUTTON_9',
    hidden: !healthChecksAdded(obj),
    href: healthChecksUrl(model, obj),
    accessReview: {
      group: model.apiGroup,
      resource: model.plural,
      name: obj.metadata.name,
      namespace: obj.metadata.namespace,
      verb: 'update',
    },
  };
};
