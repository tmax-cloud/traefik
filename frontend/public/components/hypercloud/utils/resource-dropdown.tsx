import * as _ from 'lodash-es';
import * as React from 'react';
import { Firehose, FirehoseResult, FirehoseResource, LoadingInline } from '@console/internal/components/utils';
import { ResourceListDropdown, SingleResourceDropdownProps, MultipleResourceDropdownProps } from './resource-list-dropdown';
import { useTranslation } from 'react-i18next';

const ResourceDropdownWrapper_: React.FC<ResourceDropdownWrapperProps> = props => {
  const { t } = useTranslation();
  const getItems = resources => {
    const items = [];
    _.each(resources, (resource, k) => {
      if (resource.loaded) {
        const kind = resource.kind;
        _.each(resource.data, item => {
          item.kind = kind;
          items.push(item);
        });
      }
    });
    return items;
  };

  //const [title, setTitle] = React.useState(props.loaded ? props.placeholder : <LoadingInline />);
  const [rows, setRows] = React.useState(props.loaded ? getItems(props.resources) : []);

  React.useEffect(() => {
    if (props.loaded) {
      //setTitle(props.placeholder);
      setRows(getItems(props.resources));
    }
  }, [props.loaded]);

  return (
    <ResourceListDropdown
      {...props}
      title={!props.loaded ? <LoadingInline /> : props.title}
      resourceList={rows} // 필수
      autocompletePlaceholder={t('COMMON:MSG_COMMON_BUTTON_FILTER_PLACEHOLDER_1')}
    />
  );
};

type ResourceDropdownWrapperProps = (SingleResourceDropdownProps | MultipleResourceDropdownProps) & {
  loaded?: boolean;
  resources?: FirehoseResult[];
};

const ResourceDropdown_: React.FC<ResourceDropdownProps> = ({ resources, ...props }) => {
  const resourcesWithIsListProp = resources.map(resource => {
    return { ...resource, isList: true };
  });
  return (
    <Firehose resources={resourcesWithIsListProp}>
      <ResourceDropdownWrapper_ {...props} />
    </Firehose>
  );
};

function areEqual(prevProps, nextProps) {
  // MEMO : methods와 idFunc 속성은 매번 다르게 인식돼서 memo비교문에서 제거함
  return _.isEqual(_.omit(prevProps, ['methods', 'idFunc', 'onChange']), _.omit(nextProps, ['methods', 'idFunc', 'onChange']));
}

export const ResourceDropdown = React.memo(ResourceDropdown_, areEqual);

export type ResourceDropdownProps = (SingleResourceDropdownProps | MultipleResourceDropdownProps) & {
  resources: FirehoseResource[];
};
