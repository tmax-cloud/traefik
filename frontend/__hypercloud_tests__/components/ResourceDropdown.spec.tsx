import * as React from 'react';
import { ResourceDropdown, ResourceDropdownProps } from '@console/internal/components/hypercloud/utils/resource-dropdown';
import { render, act } from '../../test-utils';
import { DeploymentModel } from '@console/internal/models';
import { DeploymentsTestData } from '../../__hypercloud_mocks__/firehoseMocks';
import * as _ from 'lodash-es';
import store from '@console/internal/redux';
import userEvent from '@testing-library/user-event';
import { configure } from '@testing-library/dom';
import { useForm, FormProvider } from 'react-hook-form';
import * as k8sActions from '@console/internal/actions/k8s';
import { inject } from '@console/internal/components/utils/inject';

const mockSubmit = jest.fn(data => {});

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: () => '' };
    return Component;
  },
}));

// MEMO : ResourceDropdown의 ResourceDropdownWrapper_가 Firehose로 감싸져있어서 Firehose컴포넌트에 대한 mocking 해줌
jest.mock('@console/internal/components/utils/firehose', () => {
  return {
    Firehose: props => {
      return inject(props.children, {
        filters: null,
        loaded: true,
        loadError: null,
        resources: {
          deployments: DeploymentsTestData,
        },
      });
    },
  };
});

const renderResourceDropdownForm = (props: ResourceDropdownProps) => {
  return render(<ResourceDropdown {...props} />, {
    wrapper: ({ children }) => {
      const methods = useForm();
      return (
        <FormProvider {...methods}>
          <form
            onSubmit={methods.handleSubmit(data => {
              mockSubmit(data);
            })}
          >
            {children}
            <button type="submit">Submit</button>
          </form>
        </FormProvider>
      );
    },
  });
};

describe('ResourceDropdown Test', () => {
  beforeEach(() => {
    const resources = {
      models: [DeploymentModel],
      adminResources: [],
      allResources: [],
      configResources: [],
      namespacedSet: null,
      safeResources: [],
      preferredVersions: [],
    };
    const action = k8sActions.receivedResources(resources);
    store.dispatch(action);
  });

  it('Single ResourceDropdown 펼쳤을 때 Deployment 리스트아이템 2개가 있어야 합니다.', () => {
    const { container, getByText, getAllByRole } = renderResourceDropdownForm({
      name: 'RD-single',
      placeholder: 'select one deployment',
      resources: [
        {
          kind: 'Deployment',
          namespace: 'catalog',
          prop: 'deployments',
        },
      ],
      type: 'single',
      useHookForm: true,
    });
    userEvent.click(getAllByRole('button')[0]);
    getByText('catalog-catalog-controller-manager');
    expect(getAllByRole('option')).toHaveLength(2);
    expect(container).toMatchSnapshot();
  });

  it('idFunc 지정 시 해당 규칙으로 form value가 생성돼야 합니다.', async () => {
    const { getByText, getAllByRole } = renderResourceDropdownForm({
      name: 'RD-single2',
      resources: [
        {
          kind: 'Deployment',
          namespace: 'catalog',
          prop: 'deployments',
        },
      ],
      type: 'single',
      useHookForm: true,
      idFunc: resource => `${resource.metadata.uid}`,
    });

    userEvent.click(getAllByRole('button')[0]);
    // MEMO : 드롭다운에서 아이템 선택 시 idFunc에 의해 리소스의 metadata.uid value가 지정이 돼야 됨
    userEvent.click(getByText('catalog-catalog-controller-manager'));
    await act(async () => userEvent.click(getByText('Submit')));
    expect(mockSubmit).toBeCalledTimes(1);
    expect(mockSubmit).toHaveBeenCalledWith({ 'RD-single2': '8b12e367-e9e3-4596-9a5c-b4884887923d' });
  });

  it('Multi ResourceDropdown 초기렌더 후 드롭다운 펼쳤을 때의 모습 테스트입니다.', () => {
    const { container, getByText, getAllByRole, getAllByText } = renderResourceDropdownForm({
      name: 'RD-multi',
      resources: [
        {
          kind: 'Deployment',
          namespace: 'catalog',
          prop: 'deployments',
        },
      ],
      type: 'multiple',
      showAll: true,
      useHookForm: true,
    });

    userEvent.click(getAllByRole('button')[0]);
    getByText('All');
    getByText('0');
    expect(getAllByText('Deployment')).toHaveLength(2);
    expect(container).toMatchSnapshot();
  });

  it('Multi ResourceDropdown 드롭다운 내 한개의 아이템 선택 시 title에 개수가 1로 표시 되어야 합니다.', () => {
    configure({ testIdAttribute: 'id' });
    const { getAllByRole, getAllByText, getByTestId } = renderResourceDropdownForm({
      name: 'RD-multi2',
      resources: [
        {
          kind: 'Deployment',
          namespace: 'catalog',
          prop: 'deployments',
        },
      ],
      type: 'multiple',
      showAll: true,
      useHookForm: true,
    });

    userEvent.click(getAllByRole('button')[0]);
    userEvent.click(getAllByText('Deployment')[0]);
    expect(getByTestId('mrd-badge')).toHaveTextContent('1');
  });

  it('Multi ResourceDropdown 드롭다운 내 모든 아이템 선택 시 All아이템항목도 체크돼야 합니다.', () => {
    configure({ testIdAttribute: 'id' });
    const { getAllByRole, getAllByText, getByTestId } = renderResourceDropdownForm({
      name: 'RD-multi3',
      resources: [
        {
          kind: 'Deployment',
          namespace: 'catalog',
          prop: 'deployments',
        },
      ],
      type: 'multiple',
      showAll: true,
      useHookForm: true,
    });

    userEvent.click(getAllByRole('button')[0]);
    getAllByText('Deployment').forEach(item => userEvent.click(item));
    expect(getByTestId('mrd-badge')).toHaveTextContent('All');
    expect(getByTestId('all-resources')).toBeChecked();
  });

  it('Multi ResourceDropdown 드롭다운 아이템리스트 검색 결과 테스트입니다.', () => {
    configure({ testIdAttribute: 'data-test-id' });
    const { getAllByRole, getAllByText, getByTestId, getByText } = renderResourceDropdownForm({
      name: 'RD-multi4',
      resources: [
        {
          kind: 'Deployment',
          namespace: 'catalog',
          prop: 'deployments',
        },
      ],
      type: 'multiple',
      showAll: true,
      useHookForm: true,
    });

    userEvent.click(getAllByRole('button')[0]);
    userEvent.type(getByTestId('dropdown-text-filter'), 'catalog-catalog-w');
    getByText('catalog-catalog-webhook');
    expect(getAllByText('Deployment')).toHaveLength(1);
  });

  it('Multi ResourceDropdown 아이템 선택 후 Submit 시 value 형태 테스트입니다.', async () => {
    const { getAllByRole, getAllByText, getByText } = renderResourceDropdownForm({
      name: 'RD-multi5',
      resources: [
        {
          kind: 'Deployment',
          namespace: 'catalog',
          prop: 'deployments',
        },
      ],
      type: 'multiple',
      showAll: true,
      useHookForm: true,
    });

    userEvent.click(getAllByRole('button')[0]);
    getAllByText('Deployment').forEach(item => userEvent.click(item));
    await act(async () => userEvent.click(getByText('Submit')));
    expect(mockSubmit).toBeCalled();
    expect(mockSubmit).toHaveBeenLastCalledWith({ 'RD-multi5': ['catalog-catalog-controller-manager', 'catalog-catalog-webhook'] });
  });
});
