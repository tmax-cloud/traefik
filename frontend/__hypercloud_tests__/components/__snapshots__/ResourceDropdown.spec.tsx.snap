// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceDropdown Test Multi ResourceDropdown 초기렌더 후 드롭다운 펼쳤을 때의 모습 테스트입니다. 1`] = `
<div>
  <form>
    <div
      class="co-type-selector"
    >
      <div
        class="dropdown pf-c-dropdown pf-m-expanded"
      >
        <button
          aria-haspopup="true"
          class="pf-c-dropdown__toggle"
          type="button"
        >
          <div
            class="btn-dropdown__content-wrap"
          >
            <span
              class="pf-c-dropdown__toggle-text"
            >
              <div>
                Resources 
                <span
                  class="pf-c-badge pf-m-read"
                  id="mrd-badge"
                >
                  0
                </span>
              </div>
            </span>
            <svg
              aria-hidden="true"
              class="pf-c-dropdown__toggle-icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 320 512"
              width="1em"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                transform=""
              />
            </svg>
          </div>
        </button>
        <ul
          class="dropdown-menu__autocomplete-filter pf-c-dropdown__menu dropdown-menu--text-wrap"
          role="listbox"
        >
          <div
            class="dropdown-menu__filter"
          >
            <input
              autocapitalize="none"
              class="pf-c-form-control"
              data-test-id="dropdown-text-filter"
              placeholder="COMMON:MSG_COMMON_BUTTON_FILTER_PLACEHOLDER_1"
              type="text"
              value=""
            />
          </div>
          <li
            class=""
            role="option"
          >
            <a
              class="pf-c-dropdown__menu-item"
              href="#"
              id="All-link"
            >
              <span
                class="co-resource-item"
              >
                <div
                  class="pf-c-check"
                >
                  <input
                    aria-invalid="false"
                    class="pf-c-check__input"
                    id="all-resources"
                    type="checkbox"
                  />
                </div>
                <span
                  class="co-resource-icon--fixed-width"
                >
                  <span
                    class="sr-only"
                  >
                    All
                  </span>
                  <span
                    class="co-m-resource-icon co-m-resource-all"
                    title="All"
                  >
                    A
                  </span>
                </span>
                <span
                  class="co-resource-item__resource-name"
                >
                  SINGLE:MSG_CONTAINERREGISTRIES_CREATEIMAGESCANREQUESTPOPUP_1
                </span>
              </span>
            </a>
          </li>
          <li
            class=""
            role="option"
          >
            <a
              class="pf-c-dropdown__menu-item"
              href="#"
              id="catalog-catalog-controller-manager-link"
            >
              <span
                class="co-resource-item"
              >
                <div
                  class="pf-c-check"
                >
                  <input
                    aria-invalid="false"
                    class="pf-c-check__input"
                    id="8b12e367-e9e3-4596-9a5c-b4884887923d:checkbox"
                    tabindex="-1"
                    type="checkbox"
                  />
                </div>
                <span
                  class="co-resource-icon--fixed-width"
                >
                  <span
                    class="sr-only"
                  >
                    Deployment
                  </span>
                  <span
                    class="co-m-resource-icon co-m-resource-deployment"
                    title="Deployment"
                  >
                    D
                  </span>
                </span>
                <span
                  class="co-resource-item__resource-name"
                >
                  <span>
                    catalog-catalog-controller-manager
                  </span>
                </span>
              </span>
            </a>
          </li>
          <li
            class=""
            role="option"
          >
            <a
              class="pf-c-dropdown__menu-item"
              href="#"
              id="catalog-catalog-webhook-link"
            >
              <span
                class="co-resource-item"
              >
                <div
                  class="pf-c-check"
                >
                  <input
                    aria-invalid="false"
                    class="pf-c-check__input"
                    id="27ddeff5-6f90-41db-8572-ddfe39face39:checkbox"
                    tabindex="-1"
                    type="checkbox"
                  />
                </div>
                <span
                  class="co-resource-icon--fixed-width"
                >
                  <span
                    class="sr-only"
                  >
                    Deployment
                  </span>
                  <span
                    class="co-m-resource-icon co-m-resource-deployment"
                    title="Deployment"
                  >
                    D
                  </span>
                </span>
                <span
                  class="co-resource-item__resource-name"
                >
                  <span>
                    catalog-catalog-webhook
                  </span>
                </span>
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`ResourceDropdown Test Single ResourceDropdown 펼쳤을 때 Deployment 리스트아이템 2개가 있어야 합니다. 1`] = `
<div>
  <form>
    <div
      class="co-type-selector"
    >
      <div
        class="dropdown pf-c-dropdown pf-m-expanded"
      >
        <button
          aria-haspopup="true"
          class="pf-c-dropdown__toggle"
          type="button"
        >
          <div
            class="btn-dropdown__content-wrap"
          >
            <span
              class="pf-c-dropdown__toggle-text"
            >
              select one deployment
            </span>
            <svg
              aria-hidden="true"
              class="pf-c-dropdown__toggle-icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 320 512"
              width="1em"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                transform=""
              />
            </svg>
          </div>
        </button>
        <ul
          class="dropdown-menu__autocomplete-filter pf-c-dropdown__menu dropdown-menu--text-wrap"
          role="listbox"
        >
          <div
            class="dropdown-menu__filter"
          >
            <input
              autocapitalize="none"
              class="pf-c-form-control"
              data-test-id="dropdown-text-filter"
              placeholder="COMMON:MSG_COMMON_BUTTON_FILTER_PLACEHOLDER_1"
              type="text"
              value=""
            />
          </div>
          <li
            class=""
            role="option"
          >
            <a
              class="pf-c-dropdown__menu-item"
              href="#"
              id="catalog-catalog-controller-manager-link"
            >
              <span
                class="co-resource-item"
              >
                <span
                  class="co-resource-icon--fixed-width"
                >
                  <span
                    class="sr-only"
                  >
                    Deployment
                  </span>
                  <span
                    class="co-m-resource-icon co-m-resource-deployment"
                    title="Deployment"
                  >
                    D
                  </span>
                </span>
                <span
                  class="co-resource-item__resource-name"
                >
                  <span>
                    catalog-catalog-controller-manager
                  </span>
                </span>
              </span>
            </a>
          </li>
          <li
            class=""
            role="option"
          >
            <a
              class="pf-c-dropdown__menu-item"
              href="#"
              id="catalog-catalog-webhook-link"
            >
              <span
                class="co-resource-item"
              >
                <span
                  class="co-resource-icon--fixed-width"
                >
                  <span
                    class="sr-only"
                  >
                    Deployment
                  </span>
                  <span
                    class="co-m-resource-icon co-m-resource-deployment"
                    title="Deployment"
                  >
                    D
                  </span>
                </span>
                <span
                  class="co-resource-item__resource-name"
                >
                  <span>
                    catalog-catalog-webhook
                  </span>
                </span>
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
